### PR TITLE
New: Add UI for subtotal requests

### PR DESCRIPTION
**Pull request by cythrawll**
## Description

Add subtotal UI buttons on column config panel to toggle rollup dimensions and grand total.

## Proposed Changes

- Create and wire up buttons to toggle rollup on dimension, datetime columns.
- Create and wire up buttons to toggle grand total.
- Buttons show up for data sources configured with the option: 'enableSubtotal'

## Screenshots
Screen Shot 2021-09-08 at 11 24 28 AM

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/consumers/request/column.ts [hunk 1]
```diff
@@ -59,6 +59,7 @@ export default class ColumnConsumer extends ActionConsumer {
       const { routeName } = route;
       const { request } = route.modelFor(routeName) as ReportModel;
       request.removeColumn(column);
+      this.requestActionDispatcher.dispatch(RequestActions.REMOVE_ROLLUP_COLUMN, route, column);
```
Comment (jkusa) [inline packages/reports/addon/consumers/request/column.ts:62]: Is this already handled here:
https://github.com/yavin-dev/framework/blob/94f19a411c94502083422b09e15792da742d2a6e/packages/reports/addon/consumers/request/rollup.ts#L56

**Comment by jkusa** [comment]
@cythrawll when the sub-total is enabled does the background change like grand total? If so, can we change the `sort` enable/disable indicator to match?

File: packages/reports/app/styles/navi-reports/components/navi-column-config.scss [hunk 2]
```diff
@@ -37,6 +60,21 @@
...
+      justify-content: space-between;
+    }
+
+    &-buttons {
+      margin-right: 4px;
+      padding: 8px 0;
+    }
+  }
+
+  &__grandtotal-icon {
```
Comment (jkusa) [inline packages/reports/app/styles/navi-reports/components/navi-column-config.scss:76]: Can we move this icon to the left a bit to align with the column arrows below?

File: packages/reports/addon/consumers/request/column.ts (hunk 1 above)
Comment (cythrawll) [reply packages/reports/addon/consumers/request/column.ts:62]: Bad rebase, i'll fix.

**Comment by cythrawll** [comment]
> @cythrawll when the sub-total is enabled does the background change like grand total? If so, can we change the `sort` enable/disable indicator to match?

Yeah i'll look at that. See screenshot for how it looks. I noticed i need to adjust the `border-radius` too.

**Comment by cythrawll** [comment]
Screen Shot 2021-09-13 at 1 41 54 PM
Screen Shot 2021-09-13 at 1 42 33 PM

**Comment by jkusa** [comment]
@cythrawll does the screenshot above indicate that if sort is not enabled, the icon is grey? If so, can we make it blue like totals?

**Comment by cythrawll** [comment]
@kilroy no, the grey indicates that sort is not available for that column. it's disabled.
the not enabled look is blue just like subtotal icon.
Screen Shot 2021-09-13 at 2 44 18 PM

File: packages/reports/addon/components/navi-column-config.ts [hunk 3]
```diff
@@ -40,26 +45,36 @@ export default class NaviColumnConfig extends Component<NaviColumnConfigArgs> {
...
       });
     }
     return [];
   }
 
+  @computed('args.report.request.dataSource')
+  get supportsSubtotal(): boolean {
+    if (!this.args.report.request.dataSource) {
+      return false;
+    }
```
Comment (jkusa) [inline packages/reports/addon/components/navi-column-config.ts:73]:
```suggestion
```
Was this for request v1? v2 requires dataSource

https://github.com/yavin-dev/framework/blob/8f7e88ef86248513080274cec9085137bc69e92f/packages/core/addon/models/bard-request-v2/request.ts#L93

**Review by jkusa — APPROVED** [review]
Just the one comment, otherwise 👍

File: packages/reports/addon/components/navi-column-config.ts (hunk 3 above)
Comment (cythrawll) [reply packages/reports/addon/components/navi-column-config.ts]: Fixed, I think I needed it before for previous code, didn't realized I coded myself out of needing this.

**Pull request merged**